### DISCUSSION TITLE
Welcome to sherpa-ncnn Discussions!

### DISCUSSION THREAD

**Discussion by csukuangfj** [k2-fsa/sherpa-ncnn#100] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by hieudevsoft** [comment]
I want implement into my project ? what should i do?

**Reply by csukuangfj** [reply]
What kind of project? C/C++/Java/Kotlin/Go/Python/C#?

**Reply by hieudevsoft** [reply]
It's  a android project with kotlin

**Reply by csukuangfj** [reply]
Please have a look at
https://github.com/k2-fsa/sherpa-onnx/blob/master/kotlin-api-examples/run.sh

Even if it is for sherpa-onnx, you can replace `onnx` with `ncnn` and it also applies to sherpa-ncnn.

**Reply by hieudevsoft** [reply]
I don't know what to do with my project?, i have one project android, and i want use feature speech recognizer in my project :(

**Reply by csukuangfj** [reply]
Could you have a look at our doc: https://k2-fsa.github.io/sherpa/ncnn/android/index.html ?

**Reply by hieudevsoft** [reply]
yes i have follow and completelly tutorial and workfine but i copy assets and jniLibs to my own project then it not working :(

**Reply by hieudevsoft** [reply]
java.lang.UnsatisfiedLinkError: No implementation found for long hieubui.eup.androidspeech.SherpaNcnn.newFromAsset(android.content.res.AssetManager, hieubui.eup.androidspeech.RecognizerConfig) (tried Java_hieubui_eup_androidspeech_SherpaNcnn_newFromAsset and Java_hieubui_eup_androidspeech_SherpaNcnn_newFromAsset__Landroid_content_res_AssetManager_2Lhieubui_eup_androidspeech_RecognizerConfig_2)
it's log

**Reply by csukuangfj** [reply]
please post the complete logs related to `sherpa-ncnn`. You can use `ncnn` as the filer in the logfiler.

**Reply by csukuangfj** [reply]
Also, please first learn the sherpa-ncnn android project and learn how sherpa-ncnn is used in Android.

**Reply by csukuangfj** [reply]
https://github.com/k2-fsa/sherpa-ncnn/blob/master/.github/scripts/test-jni.sh

This script shows how to use sherpa-ncnn in kotlin and it is not related to android.

**Comment by temcgraw** [comment]
Words are detected in silence. In microphone example, if the samples provided to s->AcceptWaveform(...) are all zero then a series of words are continuously detected. This doesn't seem to happen in the unmodified example, but if I use another audio library (miniaudio) the noise is lower and samples are closer to zero in silence.

**Reply by temcgraw** [reply]
To see the issue make the RecordCallback be:
   std::vector<float> test(frames_per_buffer, 0.0f);
   s->AcceptWaveform(16000, reinterpret_cast<const float *>(test.data()), frames_per_buffer);
   return paContinue;

**Comment by ailoli1314** [comment]


**Reply by csukuangfj** [reply]
Sorry, the code is hard to read. Could you reformat it or could you upload your python script as an attachment?

**Reply by ailoli1314** [reply]
thanks，i had see the help for Endpointing,now i know how to use this

**Comment by acjf** [comment]
我在很早的版本体验还不错。但用最近的代码体验很差，相差真的太大了。以前2022-12-17-mixed-english-chinese-sherpa-ncnn.apk的代码版本还能支持吗，我很喜欢这个的体验

**Comment by csukuangfj** [comment]
信息有限，无法提供帮助

**Comment by acjf** [comment]
https://huggingface.co/csukuangfj/sherpa-ncnn-apk/blob/main/2022-12-17-mixed-english-chinese-sherpa-ncnn.apk我想要产出这个apk的这套代码。，在这个之后的都不行，体验都很差，只有这个的很流畅

**Comment by csukuangfj** [comment]
信息有限，无法提供帮助

什么叫不行？什么叫体验很差？能否具体描述？

**Comment by acjf** [comment]
1、下载后面的版本进行编译so,这步是正常能编译的
2、进行android示例，把sherpa-ncnn-conv-emformer-transducer-2022-12-06的模型放进去
3、运行示例，基本都是会发生崩溃，无法使用

**Reply by csukuangfj** [reply]
最后一遍了, 信息有限，无法提供帮助

**Reply by acjf** [reply]
1、下载2.1.15源码
2、执行build-android-arm64-v8a.sh,得到so，拷贝android 示例的arm64-v8a目录
3、拷贝模型放到assets文件夹
4、运行，闪退：日志：
Accessing hidden method Landroid/view/View;->computeFitSystemWindows(Landroid/graphics/Rect;Landroid/graphics/Rect;)Z (greylist, reflection, allowed)
Accessing hidden method Landroid/view/ViewGroup;->makeOptionalFitsSystemWindows()V (greylist, reflection, allowed)
Start to initialize model
------config------
RecognizerConfig(feat_config=FeatureExtractorConfig(sampling_rate=16000, feature_dim=80), model_config=ModelConfig(encoder_param="sherpa-ncnn-conv-emformer-transducer-2022-12-06/encoder_jit_trace-pnnx.ncnn.int8.param", encoder_bin="sherpa-ncnn-conv-emformer-transducer-2022-12-06/encoder_jit_trace-pnnx.ncnn.int8.bin", decoder_param="sherpa-ncnn-conv-emformer-transducer-2022-12-06/decoder_jit_trace-pnnx.ncnn.param", decoder_bin="sherpa-ncnn-conv-emformer-transducer-2022-12-06/decoder_jit_trace-pnnx.ncnn.bin", joiner_param="sherpa-ncnn-conv-emformer-transducer-2022-12-06/joiner_jit_trace-pnnx.ncnn.int8.param", joiner_bin="sherpa-ncnn-conv-emformer-transducer-2022-12-06/joiner_jit_trace-pnnx.ncnn.int8.bin", tokens="sherpa-ncnn-conv-emformer-transducer-2022-12-06/tokens.txt", encoder num_threads=1, decoder num_threads=1, joiner num_threads=1), decoder_config=DecoderConfig(method="greedy_search", num_active_paths=4), endpoint_config=EndpointConfig(rule1=EndpointRule(must_contain_nonsilence=False
RecognizerConfig(feat_config=FeatureExtractorConfig(sampling_rate=16000, feature_dim=80), model_config=ModelConfig(encoder_param="sherpa-ncnn-conv-emformer-transducer-2022-12-06/encoder_jit_trace-pnnx.ncnn.int8.param", encoder_bin="sherpa-ncnn-conv-emformer-transducer-2022-12-06/encoder_jit_trace-pnnx.ncnn.int8.bin", decoder_param="sherpa-ncnn-conv-emformer-transducer-2022-12-06/decoder_jit_trace-pnnx.ncnn.param", decoder_bin="sherpa-ncnn-conv-emformer-transducer-2022-12-06/decoder_jit_trace-pnnx.ncnn.bin", joiner_param="sherpa-ncnn-conv-emformer-transducer-2022-12-06/joiner_jit_trace-pnnx.ncnn.int8.param", joiner_bin="sherpa-ncnn-conv-emformer-transducer-2022-12-06/joiner_jit_trace-pnnx.ncnn.int8.bin", tokens="sherpa-ncnn-conv-emformer-transducer-2022-12-06/tokens.txt", encoder num_threads=1, decoder num_threads=1, joiner num_threads=1), decoder_config=DecoderConfig(method="greedy_search", num_active_paths=4), endpoint_config=EndpointConfig(rule1=EndpointRule(must_contain_nonsilence=False, min_trailing_sile
AAssetManager_open sherpa-ncnn-conv-emformer-transducer-2022-12-06/encoder_jit_trace-pnnx.ncnn.int8.param failed
Failed to load sherpa-ncnn-conv-emformer-transducer-2022-12-06/encoder_jit_trace-pnnx.ncnn.int8.param
Fatal signal 11 (SIGSEGV), code 1 (SEGV_MAPERR), fault addr 0x0 in tid 13896 (fsa.sherpa.ncnn), pid 13896 (fsa.sherpa.ncnn)
pid: 13896, tid: 13896, name: fsa.sherpa.ncnn  >>> com.k2fsa.sherpa.ncnn <<<
---------------------------- PROCESS STARTED (13928) for package com.k2fsa.sherpa.ncnn ----------------------------
---------------------------- PROCESS ENDED (13885) for package com.k2fsa.sherpa.ncnn ----------------------------
#00 pc 0000000000077888  /data/app/~~PrGNfLChBdC4ICk_-eIt6Q==/com.k2fsa.sherpa.ncnn-wvVapGhciYYYbRlJz99jsQ==/lib/arm64/libsherpa-ncnn-jni.so (BuildId: f08eec7c2f0f2a56d7576ad2d3a166718938bb8e)
#01 pc 000000000005b8f4  /data/app/~~PrGNfLChBdC4ICk_-eIt6Q==/com.k2fsa.sherpa.ncnn-wvVapGhciYYYbRlJz99jsQ==/lib/arm64/libsherpa-ncnn-jni.so (BuildId: f08eec7c2f0f2a56d7576ad2d3a166718938bb8e)
#02 pc 000000000005a104  /data/app/~~PrGNfLChBdC4ICk_-eIt6Q==/com.k2fsa.sherpa.ncnn-wvVapGhciYYYbRlJz99jsQ==/lib/arm64/libsherpa-ncnn-jni.so (Java_com_k2fsa_sherpa_ncnn_SherpaNcnn_newFromAsset+336) (BuildId: f08eec7c2f0f2a56d7576ad2d3a166718938bb8e)
#10 pc 0000000000003a02  [anon:dalvik-classes3.dex extracted in memory from /data/app/~~PrGNfLChBdC4ICk_-eIt6Q==/com.k2fsa.sherpa.ncnn-wvVapGhciYYYbRlJz99jsQ==/base.apk!classes3.dex] (com.k2fsa.sherpa.ncnn.SherpaNcnn.<init>+30)
#13 pc 0000000000001cda  [anon:dalvik-classes3.dex extracted in memory from /data/app/~~PrGNfLChBdC4ICk_-eIt6Q==/com.k2fsa.sherpa.ncnn-wvVapGhciYYYbRlJz99jsQ==/base.apk!classes3.dex] (com.k2fsa.sherpa.ncnn.MainActivity.initModel+130)
#16 pc 0000000000001d4e  [anon:dalvik-classes3.dex extracted in memory from /data/app/~~PrGNfLChBdC4ICk_-eIt6Q==/com.k2fsa.sherpa.ncnn-wvVapGhciYYYbRlJz99jsQ==/base.apk!classes3.dex] (com.k2fsa.sherpa.ncnn.MainActivity.onCreate+50)
---------------------------- PROCESS ENDED (13896) for package com.k2fsa.sherpa.ncnn ----------------------------

**Reply by acjf** [reply]
每一个tag的版本，我均尝试过，都是不同的表现，这里仅提供了最新版本的崩溃日志

**Reply by acjf** [reply]
模型使用的是：https://k2-fsa.github.io/sherpa/ncnn/pretrained_models/conv-emformer-transducer-models.html#csukuangfj-sherpa-ncnn-conv-emformer-transducer-2022-12-06-chinese-english 下载的模型

**Comment by csukuangfj** [comment]
感谢分享详细的 log! 我们现在可以提供帮助了。


> AAssetManager_open sherpa-ncnn-conv-emformer-transducer-2022-12-06/encoder_jit_trace-pnnx.ncnn.int8.param failed
Failed to load sherpa-ncnn-conv-emformer-transducer-2022-12-06/encoder_jit_trace-pnnx.ncnn.int8.param

这个是关键的 log

请把你 assets 文件夹，在 android stduio 里，截个图，看里面有什么。

**Reply by acjf** [reply]
<img width="960" height="1286" alt="wechat_2025-09-26_155912_938" src="https://github.com/user-attachments/assets/40d5a3f7-53de-486c-a3e5-92de89df9e04" />

**Reply by acjf** [reply]
我看到问题，我试修改下

**Reply by csukuangfj** [reply]
> 我看到问题，我试修改下

太好啦！你终于会自己调试了

**Reply by acjf** [reply]
很抱歉，是我的问题，我犯了一个低级的错误。感谢你的耐心

**Reply by csukuangfj** [reply]
如果你觉得有用，请给我们点 star, 感谢支持!

**Reply by acjf** [reply]
咱们的推理框架支持耳语的模型吗，或者说耳语的能转成ncnn所需的模型吗

**Reply by csukuangfj** [reply]
如前所述，信息有限，无法回答。

**Comment by csukuangfj** [comment]
友情提示，如果你用最新的代码，只需要 libncnn.so 和 libsherpa-ncnn-jni.so, 其他的 .so 和 .a 文件，都可以删掉。

**Comment by sys-dotcom** [comment]
我现在有个需求，是做一个本地化部署的中英文语音翻译，就是讲话时，可以实时出现对照的中英文，并且可以降噪，不知道这个是否可以支持？如果是可以的，可以进行合作

**Comment by sys-dotcom** [comment]
目前，我这有大量可以支撑训练的语料，之前用了科大讯飞的，但是效果不理想，识别率不足90%，因为语料很多是船舶行业的专有名词，所以，这个语言模型是否可以支持训练？
